cuqi/solver/_solver.py

```diff
@@ -320,6 +326,137 @@ def solve(self):
         return x, k
 
 
+class PCGLS:
+    """Conjugate Gradient method for least squares problems with preconditioning
+    
+    See Bjorck (1996) - Numerical Methods for Least Squares Problems. Pag 294.
+
+    Parameters
+    ----------
+    A : ndarray or callable f(x,*args).
+    b : ndarray.
+    x0 : ndarray. Initial guess.
+    P : ndarray. Preconditioner array in sparse format.
+    maxit : The maximum number of iterations.
+    tol : The numerical tolerance for convergence checks.
```

Comment:
Could you add a few more comments here explaning what the parameters are. Specifically what is A, b and P. Show equation that is being solved.

Also please use the same format as in e.g. the [L_BGFS_B](https://github.com/CUQI-DTU/CUQIpy/blob/main/cuqi/solver/_solver.py#L17) class for parameters:
```
argument : type
    Description
```